storage/factory/pruningStorerFactory.go

```diff
@@ -246,6 +247,17 @@ func (psf *StorageServiceFactory) CreateForShard() (dataRetriever.StorageService
 		return nil, err
 	}
 
+	if psf.generalConfig.StoragePruning.CleanOldEpochsData {
+		_, err = olddbclean.NewOldDatabaseCleaner(olddbclean.Args{
```

Comment:
why do we call the New method but we do not keep the pointer? Valid also for CreateForMeta() method
L.E. : because it just registers to the epoch notifier. My suggestion is to keep the pointer somewhere in case we need to call the Close method sometimes.

Reply:
I didn't know where to keep it. But after system tests, the GC realizes that the struct is in use and won't delete it. Any suggestion of where to store it?

Reply:
a new field in StorageService implementations? I do not like pointers created but held only by the runtime automatically, our code being unable to retriever it. What do you think, @AdoAdoAdo ?

Reply:
I also agree that it looks a bit weird.
But I would rather not change the storage service if there is no need to do extra operations to close this component.
The pointer is added to the subscribers list to be notified on epoch change events, and a close should clean it up there on the epoch change notifier.